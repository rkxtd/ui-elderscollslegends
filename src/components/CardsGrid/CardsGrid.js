import React from "react";
import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import Alert from "@material-ui/lab/Alert";

import PictureCard from "../PictureCard";

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: 76,
    flexGrow: 1,
    maxWidth: 1280,
    margin: "auto",
  },
  error: {
    width: "100%",
    padding: 20,
  },
  paper: {
    height: 340,
    width: 260,
  },
  control: {
    padding: theme.spacing(2),
  },
}));

export default function CardsGrid({ cards, errorMessage }) {
  const classes = useStyles();
  return (
    <Grid container className={classes.root}>
      {errorMessage && (
        <Alert className={classes.error} severity="error">
          {errorMessage}
        </Alert>
      )}
      <Grid item xs={12}>
        <Grid container justify="center">
          {cards.map(
            ({
              id,
              imageUrl,
              name: cardName,
              type: cardType,
              text: cardText,
              set: { name: setName },
            }) => (
              <Grid key={id} item>
                <PictureCard
                  className={classes.paper}
                  imageUrl={imageUrl}
                  cardName={cardName}
                  cardText={cardText}
                  cardType={cardType}
                  setName={setName}
                />
              </Grid>
            )
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}
