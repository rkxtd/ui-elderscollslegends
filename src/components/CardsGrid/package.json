{
  "main": "CardsGrid.js",
  "devDependencies": {
    "@babel/core": "^7.12.1"
  }
}
