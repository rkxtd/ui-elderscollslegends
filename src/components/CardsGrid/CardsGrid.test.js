import React from "react";
import renderer from 'react-test-renderer';
import { describe, it } from "@jest/globals";
import { shallow } from "enzyme";
import CardsGrid from "./CardsGrid";

const cards = [{
  id: 'card1',
  imageUrl: '',
  name: 'Card 1',
  type: 'Type 1',
  text: 'Text on Card 1',
  set: { name: 'Set 1' },
}, {
  id: 'card2',
  imageUrl: '',
  name: 'Card 2',
  type: 'Type 1',
  text: 'Text on Card 2',
  set: { name: 'Set 1' },
}, {
  id: 'card3',
  imageUrl: '',
  name: 'Card 3',
  type: 'Type 2',
  text: 'Text on Card 3',
  set: { name: 'Set 1' },
}];

describe("Test CardsGrid Component", () => {
  describe("Test shallow component", () => {
    it("renders empty grid without crashing.", () => {
      shallow(<CardsGrid cards={[]} />);
    });

    it("renders grid with cards.", () => {
      shallow(<CardsGrid cards={cards} />);
    });

    it("renders grid with cards and error message.", () => {
      shallow(<CardsGrid cards={cards} errorMessage={"API ERROR"} />);
    });
  });

  describe("Test Snapshot", () => {
    it('renders without cards.', () => {
      const tree = renderer
        .create(<CardsGrid cards={[]} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders with cards.', () => {
      const tree = renderer
        .create(<CardsGrid cards={cards} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders with cards and error message.', () => {
      const tree = renderer
        .create(<CardsGrid cards={cards} errorMessage={"API ERROR"} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  })
});
