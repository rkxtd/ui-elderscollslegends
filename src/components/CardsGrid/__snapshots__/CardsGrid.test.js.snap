// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardsGrid Component Test Snapshot renders with cards and error message. 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiPaper-root MuiAlert-root MuiAlert-standardError makeStyles-error-2 MuiPaper-elevation0"
    role="alert"
  >
    <div
      className="MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message"
    >
      API ERROR
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 1"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 1"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 1
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 1
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 1
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 2"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 2"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 2
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 1
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 2
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 3"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 3"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 3
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 2
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test CardsGrid Component Test Snapshot renders with cards. 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 1"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 1"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 1
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 1
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 1
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 2"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 2"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 2
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 1
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 2
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Card 3"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="560"
            title="Card 3"
          />
          <div
            className="MuiCardContent-root makeStyles-content-6"
          >
            <h3
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Card 3
            </h3>
            <h4
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Type 2
               / 
              Set 1
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Text on Card 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test CardsGrid Component Test Snapshot renders without cards. 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    />
  </div>
</div>
`;
